PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/helpers/deploy/KuduClient.kt

```diff
@@ -46,15 +47,35 @@ object KuduClient {
     /**
      * Method to publish specified ZIP file to Azure server. We make up to 3 tries for uploading a ZIP file.
      *
-     * Note: Azure SDK support a native [FunctionApp.zipDeploy(File)] method. Hoverer, we cannot use it for files with BOM
+     * Note: Azure SDK supports a native [FunctionApp.zipDeploy(File)] method. Hoverer, we cannot use it for files with BOM
      *       Method throws an exception while reading the JSON file that contains BOM. Use our own implementation until fixed
      *
      * @param zipFile - zip file instance to be published
      * @param processHandler - a process handler to show a process message
      *
      * @throws [RuntimeException] in case REST request was not succeed or timed out after 3 attempts
      */
-    fun kuduZipDeploy(zipFile: File, publishingProfile: PublishingProfile, appName: String, processHandler: RunProcessHandler) {
+    fun kuduZipDeploy(zipFile: File, publishingProfile: PublishingProfile, appName: String, kuduWebApp: WebAppBase, processHandler: RunProcessHandler) {
```

Comment:
I think you can simplify this overload since you pass a WebAppBase object:
```
fun kuduZipDeploy(zipFile: File, app: WebAppBase, processHandler: RunProcessHandler) {
    val appName = app.name()
    val kuduBaseUrl = "https://" + app.defaultHostName().toLowerCase()
	                .replace("http://", "")
	                .replace(appName.toLowerCase(), appName.toLowerCase() + ".scm")
   
   kuduZipDeploy(zipFile, app.publishingProfile, appName, kuduBaseUrl, processHandler)
}
```